testsuite/smoke-test/src/rosetta.rs

```diff
@@ -54,36 +63,111 @@ pub async fn setup_test(
 }
 
 #[tokio::test]
-#[ignore]
+async fn test_network() {
+    let (swarm, _, _, rosetta_client) = setup_test(1, 1).await;
+    let chain_id = swarm.chain_id();
+
+    // We only support one network, this network
+    let networks = rosetta_client.network_list().await.unwrap();
+    assert_eq!(1, networks.network_identifiers.len());
+    let network_id = networks.network_identifiers.first().unwrap();
+    assert_eq!(BLOCKCHAIN, network_id.blockchain);
+    assert_eq!(chain_id.to_string(), network_id.network);
+
+    let request = NetworkRequest {
+        network_identifier: NetworkIdentifier::from(chain_id),
+    };
+    let options = rosetta_client.network_options(&request).await.unwrap();
+    assert_eq!(ROSETTA_VERSION, options.version.rosetta_version);
+
+    // TODO: Check other options
+
+    let request = NetworkRequest {
+        network_identifier: NetworkIdentifier::from(chain_id),
+    };
+    let status = rosetta_client.network_status(&request).await.unwrap();
+    assert!(status.current_block_identifier.index > 0);
+    assert!(status.current_block_timestamp > Y2K_SECS);
+    assert_eq!(
+        BlockIdentifier {
+            index: 0,
+            hash: HashValue::zero().to_hex()
+        },
+        status.genesis_block_identifier
+    );
+    assert_eq!(
+        Some(status.genesis_block_identifier),
+        status.oldest_block_identifier,
+    );
+}
+
+#[tokio::test]
 async fn test_account_balance() {
     let (swarm, cli, _faucet, rosetta_client) = setup_test(1, 1).await;
 
     cli.create_account_with_faucet(0).await.unwrap();
     let account = CliTestFramework::account_id(0);
     let chain_id = swarm.chain_id();
+
+    // At time 0, there should be 0 balance
+    let response = get_balance(&rosetta_client, chain_id, account, 0)
+        .await
+        .unwrap();
+    assert_eq!(
+        response.block_identifier,
+        BlockIdentifier {
+            index: 0,
+            hash: HashValue::zero().to_hex(),
+        }
+    );
+
+    // At some time before version 100, the account should exist
+    let mut successful_version = None;
```

Comment:
What's the timing here? Waiting for the account creation request to be created? Hence we wait for up to 100 versions because there could be some block metadata transactions (for example) beforehand?

Might be worth bumping this to like 1000 to make sure it's not flaky.

Reply:
Yeah, probably can update to 1000, though we don't process 1000 blocks that fast in the testing environment